projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts

```diff
@@ -69,7 +69,7 @@ export class NgxIndexedDBService {
    * @param value The entry to be added
    * @param key The optional key for the entry
    */
-  add<T>(storeName: string, value: T, key?: any): Observable<T> {
+  add<T>(storeName: string, value: T, key?: any): Observable<T & {id: any}> {
```

Comment:
Let's create `type WithID = {}` and reuse it :)

Reply:
Great idea. I have pushed new update with new type (WithID)